update/update-version.md

```diff
@@ -86,25 +86,44 @@ If you have both the code and the Terraform state file stored locally on your ma
     ```sh
-        export IBMCLOUD_API_KEY="<API-KEY>" #pragma: allowlist secret
+    export IBMCLOUD_API_KEY="<API-KEY>" #pragma: allowlist secret
     ```
 
-1. Retrive the VPC IDs of the VPC on which all the VSIs are deployed, including the region of VPC.
-    1. VPC ID can be retrieved from the terraform output by running `terraform output`.
+1. Retrive the VPC ID of the VPC on which all the VSIs are deployed, including the region of VPC.
+    1. VPC ID can be retrieved from the terraform output by running `terraform output` or use the following command to get the VPC ID
+    ```sh
+    terraform output -json | jq -r '.. | objects | .value' | jq -r '.. | objects | select(.vpc_id != null) | .vpc_id' | sort -u
+    ```
+
+1. Download/Copy the script to the same directory in which the state file exists.
+```sh
+curl https://raw.githubusercontent.com/terraform-ibm-modules/terraform-ibm-landing-zone-vsi/refactor/update/update_v3.x.x_to_v4.x.x.sh > update_v3.x.x_to_v4.x.x.sh
+```
 
 1. Run the script
 ```sh
 bash update_v3.x.x_to_v4.x.x.sh -v "<vpc-id1>,[<vpc-id2>,...]" -r "<vpc-region>"
 ```
-
-1. Now pull in the latest release and run `terraform plan` and make sure none of the VSIs will be recreated.
-
-1. Run `terraform apply`.
-
-1. [Optional] If an issue occurs during the Terraform state migration, you can undo the modifications made by the script by running it again with the -z option. For example,
+In the unlikely event that an issue occurs, you can undo the modifications made by the script by running it again with the -z option. For example,
 ```sh
 bash update_v3.x.x_to_v4.x.x.sh -z
 ```
 Reverting the Terraform state file back to its prior condition, which existed prior to the execution of the script.
+
+**Note:** If the script did fail, once the user reverts, they should not proceed with remaining steps and instead create an IBM Cloud support case and reference the script along with the error.
+
+1. Now pull in the latest release and run `terraform plan` and make sure none of the VSIs will be recreated.
+    1. Update the version of the module in your consuming code to the 4.x.x version. For example:\
+        ```hcl
+        source                           = "terraform-ibm-modules/landing-zone-vsi/ibm"
+        version                          = "4.0.0"
+        ```
+    1. Run terraform init to pull the latest version
+    1. Run terraform plan to ensure none of the VSIs will be recreated. You should see an update in-place for all of the VSIs in your state. The update will rename the VSI to the new naming convention of prefix + last 4 digits of subnet_id + padding per subnet. For example:
+    ```sh
+    ~ name                             = "prefix-001" -> "prefix-3ad7-001"
+    ```
+
+1. Run `terraform apply` to upgrade to the 4.x.x version of the module and apply the update in place to rename the VSIs.
```

Comment:
I see the script created `moved.txt` and `revert.txt` in the same directory as the script was run. I was going to suggest using a temp dir, but I guess you need to know the exact location and name of the file incase a revert is needed right?
So maybe the best we can do is document that these files can be manually cleaned after successful upgrade.